### ISSUE TITLE
OutOfMemoryError "Could not allocate JNI Env"

### ISSUE THREAD

**Issue by OleksandrSlobodchukov** [andriydruk/RxDNSSD#11]
```
W/art: Throwing OutOfMemoryError "Could not allocate JNI Env"
E/AndroidRuntime: FATAL EXCEPTION: Thread-15287
    Process: com.livestream.mevo, PID: 31626
    java.lang.OutOfMemoryError: Could not allocate JNI Env
        at java.lang.Thread.nativeCreate(Native Method)
        at java.lang.Thread.start(Thread.java:1063)
        at com.apple.dnssd.AppleQuery.<init>(DNSSD.java:844)
        at com.apple.dnssd.AppleDNSSD._queryRecord(DNSSD.java:591)
        at com.apple.dnssd.DNSSD.queryRecord(DNSSD.java:321)
        at com.github.druk.rxdnssd.RxDnssdCommon$3$1$2.getService(RxDnssdCommon.java:108)
        at com.github.druk.rxdnssd.RxDnssdBindable$DNSSDServiceAction.call(RxDnssdBindable.java:65)
        at com.github.druk.rxdnssd.RxDnssdBindable$DNSSDServiceAction.call(RxDnssdBindable.java:51)
        at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:48)
        at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:30)
        at rx.Observable.unsafeSubscribe(Observable.java:10150)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.onNext(OperatorMerge.java:250)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.onNext(OperatorMerge.java:147)
        at rx.internal.operators.OnSubscribeFromArray$FromArrayProducer.fastPath(OnSubscribeFromArray.java:76)
        at rx.internal.operators.OnSubscribeFromArray$FromArrayProducer.request(OnSubscribeFromArray.java:58)
        at rx.Subscriber.setProducer(Subscriber.java:211)
        at rx.internal.operators.OnSubscribeFromArray.call(OnSubscribeFromArray.java:32)
        at rx.internal.operators.OnSubscribeFromArray.call(OnSubscribeFromArray.java:24)
        at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:48)
        at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:30)
        at rx.Observable.unsafeSubscribe(Observable.java:10150)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.onNext(OperatorMerge.java:250)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.onNext(OperatorMerge.java:147)
        at rx.internal.operators.OnSubscribeMap$MapSubscriber.onNext(OnSubscribeMap.java:77)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.emitLoop(OperatorMerge.java:732)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.emitScalar(OperatorMerge.java:437)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.tryEmit(OperatorMerge.java:357)
        at rx.internal.operators.OperatorMerge$InnerSubscriber.onNext(OperatorMerge.java:852)
        at rx.observers.Subscribers$5.onNext(Subscribers.java:235)
        at com.github.druk.rxdnssd.RxResolveListener.serviceResolved(RxResolveListener.java:46)
        at com.apple.dnssd.AppleService.ProcessResults(Native Method)
        at com.apple.dnssd.AppleService.run(DNSSD.java:693)
        at java.lang.Thread.run(Thread.java:818)
```

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/142341493853498/224288599827186)

**Comment by lrannn** [comment]
Maybe it's the same problem. HUAWEI device, android 7.0.

```
java.lang.OutOfMemoryError: pthread_create (1040KB stack) failed: Out of memory
	at java.lang.Thread.nativeCreate(Native Method)
	at java.lang.Thread.start(Thread.java:745)
	at com.apple.dnssd.AppleQuery.<init>(DNSSD.java:844)
	at com.apple.dnssd.AppleDNSSD._queryRecord(DNSSD.java:591)
	at com.apple.dnssd.DNSSD.queryRecord(DNSSD.java:321)
	at com.github.druk.rxdnssd.RxDnssdCommon$3$1$1.getService(RxDnssdCommon.java:111)
	at com.github.druk.rxdnssd.RxDnssdBindable$DNSSDServiceAction.call(RxDnssdBindable.java:67)
	at com.github.druk.rxdnssd.RxDnssdBindable$DNSSDServiceAction.call(RxDnssdBindable.java:53)
	at rx.Observable$2.call(Observable.java:233)
	at rx.Observable$2.call(Observable.java:225)
	at rx.Observable.unsafeSubscribe(Observable.java:8741)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.onNext(OperatorMerge.java:245)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.onNext(OperatorMerge.java:145)
	at rx.internal.operators.OnSubscribeFromArray$FromArrayProducer.fastPath(OnSubscribeFromArray.java:76)
	at rx.internal.operators.OnSubscribeFromArray$FromArrayProducer.request(OnSubscribeFromArray.java:58)
	at rx.Subscriber.setProducer(Subscriber.java:211)
	at rx.internal.operators.OnSubscribeFromArray.call(OnSubscribeFromArray.java:32)
	at rx.internal.operators.OnSubscribeFromArray.call(OnSubscribeFromArray.java:24)
	at rx.Observable$2.call(Observable.java:233)
	at rx.Observable$2.call(Observable.java:225)
	at rx.Observable.unsafeSubscribe(Observable.java:8741)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.onNext(OperatorMerge.java:245)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.onNext(OperatorMerge.java:145)
	at rx.internal.operators.OperatorMap$1.onNext(OperatorMap.java:54)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.emitLoop(OperatorMerge.java:705)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.emitScalar(OperatorMerge.java:416)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.tryEmit(OperatorMerge.java:340)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.onNext(OperatorMerge.java:825)
	at rx.observers.Subscribers$5.onNext(Subscribers.java:229)
	at com.github.druk.rxdnssd.RxResolveListener.serviceResolved(RxResolveListener.java:44)
	at com.apple.dnssd.AppleService.ProcessResults(Native Method)
	at com.apple.dnssd.AppleService.run(DNSSD.java:693)
	at java.lang.Thread.run(Thread.java:776)
```

**Comment by connected-jallen** [comment]
java.lang.OutOfMemoryError: Could not allocate JNI Env

FRAME	MODULE					CRASH DETAILS
0	java​.lang					Thread.nativeCreate
1	java​.lang					Thread.java ​ Line 1063
								Thread.start
2	com​.apple​.dnssd				DNSSD.java ​ Line 844
								AppleQuery.<init>
3	com​.apple​.dnssd				DNSSD.java ​ Line 591
								AppleDNSSD._queryRecord
4	com​.apple​.dnssd				DNSSD.java ​ Line 321
								DNSSD.queryRecord
5	com​.github​.druk​.rxdnssd		RxDnssdCommon.java ​ Line 108
								RxDnssdCommon$3$1$2.getService
6	com​.github​.druk​.rxdnssd		RxDnssdBindable.java ​ Line 65
								RxDnssdBindable$DNSSDServiceAction.call
7	com​.github​.druk​.rxdnssd		RxDnssdBindable.java ​ Line 51
								RxDnssdBindable$DNSSDServiceAction.call
8	rx​.internal​.operators		OnSubscribeLift.java ​ Line 48
								OnSubscribeLift.call
9	rx​.internal​.operators		OnSubscribeLift.java ​ Line 30
								OnSubscribeLift.call
10	rx							Observable.java ​ Line 10144
								Observable.unsafeSubscribe
11	rx​.internal​.operators		OperatorMerge.java ​ Line 248
								OperatorMerge$MergeSubscriber.onNext
12	rx​.internal​.operators		OperatorMerge.java ​ Line 148
								OperatorMerge$MergeSubscriber.onNext
13	rx​.internal​.operators		OnSubscribeFromArray.java ​ Line 76
								OnSubscribeFromArray$FromArrayProducer.fastPath
14	rx​.internal​.operators		OnSubscribeFromArray.java ​ Line 58
								OnSubscribeFromArray$FromArrayProducer.request
15	rx							Subscriber.java ​ Line 211
								Subscriber.setProducer
16	rx​.internal​.operators		OnSubscribeFromArray.java ​ Line 32
								OnSubscribeFromArray.call
17	rx​.internal​.operators		OnSubscribeFromArray.java ​ Line 24
								OnSubscribeFromArray.call
18	rx​.internal​.operators		OnSubscribeLift.java ​ Line 48
								OnSubscribeLift.call
19	rx​.internal​.operators		OnSubscribeLift.java ​ Line 30
								OnSubscribeLift.call
20	rx							Observable.java ​ Line 10144
								Observable.unsafeSubscribe
21	rx​.internal​.operators		OperatorMerge.java ​ Line 248
								OperatorMerge$MergeSubscriber.onNext
22	rx​.internal​.operators		OperatorMerge.java ​ Line 148
								OperatorMerge$MergeSubscriber.onNext
23	rx​.internal​.operators		OnSubscribeMap.java ​ Line 77
								OnSubscribeMap$MapSubscriber.onNext
24	rx​.internal​.operators		OperatorMerge.java ​ Line 395
								OperatorMerge$MergeSubscriber.emitScalar
25	rx​.internal​.operators		OperatorMerge.java ​ Line 355
								OperatorMerge$MergeSubscriber.tryEmit
26	rx​.internal​.operators		OperatorMerge.java ​ Line 846
								OperatorMerge$InnerSubscriber.onNext
27	rx​.observers				Subscribers.java ​ Line 235
								Subscribers$5.onNext
28	com​.github​.druk​.rxdnssd		RxResolveListener.java ​ Line 46
								RxResolveListener.serviceResolved
29	com​.apple​.dnssd				AppleService.ProcessResults
30	com​.apple​.dnssd				DNSSD.java ​ Line 693
								AppleService.run
31	java​.lang					Thread.java ​ Line 818
								Thread.run

**Comment by andriydruk** [comment]
Fixed in [0.9.3](https://github.com/andriydruk/RxDNSSD/releases/tag/0.9.3)
